syftlib/src/test/java/org/openmined/syft/unit/monitor/network/NetworkValidityTest.kt

```diff
@@ -0,0 +1,109 @@
+import org.openmined.syft.monitor.network.NetworkStatusModel
+import org.openmined.syft.monitor.network.NetworkStatusRealTimeDataSource
+import org.openmined.syft.monitor.network.NetworkStatusRepository
+
+@ExperimentalUnsignedTypes
+class NetworkValidityTest : AbstractSyftWorkerTest() {
+
+    private lateinit var config: SyftConfiguration
+
+    @Before
+    fun setUp() {
+        val socketClient = SocketClientMock(
+            authenticateSuccess = true,
+            cycleSuccess = true
+        )
+
+        val httpClient = HttpClientMock(
+            pingSuccess = true, downloadSpeedSuccess = true,
+            uploadSuccess = true, downloadPlanSuccess = true, downloadModelSuccess = true
+        )
+
+        config = SyftConfiguration(
+            context,
+            networkingSchedulers,
+            computeSchedulers,
+            context.filesDir,
+            true,
+            networkConstraints,
+            NetworkCapabilities.TRANSPORT_WIFI,
+            0,
+            1,
+            socketClient.getMockedClient(),
+            httpClient.getMockedClient(),
+            SyftConfiguration.NetworkingClients.HTTP
+        )
+
+    }
+
+    // Test for NetworkStatusRealTimeDataSource's getNetworkValidity
+    @Test
+    fun `network validity should return true if network constraints satisfied`() {
+        val networkDataSource = NetworkStatusRealTimeDataSource.initialize(config)
+        assert(networkDataSource.getNetworkValidity(networkConstraints))
+    }
+
+
+    // Test for NetworkRepository's getNetworkValidity
+    @Test
+    fun `network validity call NetworkStatusDataSource's getNetworkValidity correctly`() {
+        val networkStatusRealDataSource = NetworkStatusRealTimeDataSource.initialize(config)
+        val networkStatusRepository = NetworkStatusRepository(listOf(), mock(), networkStatusRealDataSource)
+        assert(networkStatusRepository.getNetworkValidity())
+    }
+
+    // Test for NetworkRepository's getNetworkStatus
+    @Test
+    fun `getNetworkStatus should return cached network status model`() {
+        val networkStatusModel = NetworkStatusModel("1", "1", "1", true)
+        val cache = mock<NetworkStatusCache> {
+            on { getNetworkStatusCache() }.thenReturn(Maybe.just(networkStatusModel))
+        }
+
+        val networkStatusRealDataSource = NetworkStatusRealTimeDataSource.initialize(config)
+        val networkStatusRepository = NetworkStatusRepository(listOf(), cache, networkStatusRealDataSource)
+        val cachedStatus = networkStatusRepository.getNetworkStatus("worker id").blockingGet()
+
+        assert(cachedStatus.ping != null)
+        assert(cachedStatus.downloadSpeed != null)
+        assert(cachedStatus.uploadspeed != null)
+        assert(cachedStatus.networkValidity)
+    }
+
+    // Test for NetworkRepository's getNetworkStatus
+    @Test
+    fun `getNetworkStatus calculate new NetworkModelStatus if there no cache`() {
```

Comment:
This test can also be done via mocked instance

Reply:
I managed to get mock the others without robolectric, but for this when calculating new model status it requires a context to get networkVaildity if I mock this function I will get null pointer error on the downloader. I'll push a draft RP